src/auth_components/Login.tsx

```diff
@@ -18,25 +22,30 @@ export default function Login({
   const [message, setMessage] = useState("")
   const [loading, setLoading] = useState(false)
 
-  async function handleSubmit(e: any) {
-    e.preventDefault()
+  const history = useHistory()
+
+  async function handleSubmit (event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault()
     try {
       setMessage("")
       setLoading(true)
       await login(emailRef.current.value, passwordRef.current.value)
-    } catch(error) {
-      console.log(error.code)
-      const userMessage = error.code === 'auth/wrong-password'? 'Wrong password, please try again.' : error.message
-      console.log(error.message)
-      setMessage(userMessage)
+      if (didClickSavedBuildings) { history.push("./saved-homes") }
```

Comment:
A slightly more decoupled way to handle this might be to have the `Login` modal accept an `afterLogin` callback. Then, this `history.push` call could just be passed in as a callback function when the `LOGIN_SAVED_BUILDINGS` state is handled, vs. passing `true` or `false` for this prop.